### PR TITLE
Fixes for several facets of the rendering.

### PR DESCRIPTION
**Pull request by marchermans**
Fixes several issues with the new rendering mechanics:
1. Rotation of the Template is now around the center of the Primary block.
2. Exclusion handlers for substitution blocks, prevents both substitution blocks from being rendered in Templates.
3. Mirroring is now on the correct axis, and implement the rendering for the axis that is not being used.

### CODE REVIEW

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/2374/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

File: src/main/java/com/minecolonies/structures/lib/RenderUtil.java [hunk 1]
```diff
@@ -19,6 +19,8 @@ private RenderUtil()
 
     public static void applyRotationToYAxis(@NotNull final Rotation rotation)
     {
+        GlStateManager.translate(0.5F, 0F, 0.5F);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/structures/lib/RenderUtil.java:22]: MAJOR Assign this magic number 0.5F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 0.5F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/structures/lib/RenderUtil.java [hunk 2]
```diff
@@ -40,6 +42,8 @@ public static void applyRotationToYAxis(@NotNull final Rotation rotation)
         }
 
         GlStateManager.rotate(angle, 0, 1, 0);
+
+        GlStateManager.translate(-0.5F, 0F, -0.5F);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/structures/lib/RenderUtil.java:46]: MAJOR Assign this magic number 0.5F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
MAJOR Assign this magic number 0.5F to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

File: src/main/java/com/minecolonies/structures/client/TemplateRenderHandler.java [hunk 3]
```diff
@@ -38,6 +45,11 @@ public static TemplateRenderHandler getInstance()
 
     public void draw(final Template template, final Rotation rotation, final Mirror mirror, final Vector3d drawingOffset)
     {
+        if (rendererDispatcher == null)
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/structures/client/TemplateRenderHandler.java]: CRITICAL Missing curly brace. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00121)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 4]
```diff
@@ -252,6 +254,10 @@ public static void registerModels(@NotNull final ModelRegistryEvent event)
             ModelLoader.setCustomModelResourceLocation(Item.getItemFromBlock(frame), 0,
                         new ModelResourceLocation(frame.getRegistryName(), INVENTORY));
         }
+
+        //Additionally we register an exclusion handler here;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java:258]: MINOR This block of commented-out lines of code should be removed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 5]
```diff
@@ -252,6 +254,10 @@ public static void registerModels(@NotNull final ModelRegistryEvent event)
...
+        TemplateRenderHandler.getInstance().registerExclusionHandler((b) -> b.blockState.getBlock() instanceof BlockSubstitution);
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java:259]: MINOR Remove the parentheses around the "b" parameter (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1611)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 6]
```diff
@@ -252,6 +254,10 @@ public static void registerModels(@NotNull final ModelRegistryEvent event)
...
+
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java]: MINOR Remove the parentheses around the "b" parameter (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1611)

File: src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java [hunk 7]
```diff
@@ -20,9 +20,11 @@
 import com.minecolonies.coremod.entity.ai.mobs.barbarians.EntityChiefBarbarian;
 import com.minecolonies.coremod.event.ClientEventHandler;
 import com.minecolonies.coremod.items.ModItems;
+import com.minecolonies.coremod.placementhandlers.PlacementHandlers;
```
Comment (BuildTeamLinter) [inline src/main/java/com/minecolonies/coremod/proxy/ClientProxy.java:23]: MINOR Remove this unused import 'com.minecolonies.coremod.placementhandlers.PlacementHandlers'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 59 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 5 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 58 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by BuildTeamLinter** [comment]
SonarQube analysis reported 57 issues
* ![MAJOR][MAJOR] 53 major
* ![MINOR][MINOR] 4 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [blocks#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/blocks#): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [gui#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/gui#): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [render#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/client/render#): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [colony#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony#): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [buildings#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/buildings#): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [permissions#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/permissions#): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [requestable#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestable#): Add a 'package-info.java' file to document the 'requestable' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [data#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/data#): Add a 'package-info.java' file to document the 'data' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [init#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/init#): Add a 'package-info.java' file to document the 'init' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MAJOR][MAJOR] [locations#](https://github.com/Minecolonies/minecolonies/blob/20558fe4aaf2b826c33563f7247648572f7b7129/src/main/java/com/minecolonies/coremod/colony/requestsystem/locations#): Add a 'package-info.java' file to document the 'locations' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Pull request merged**